dra-evolution/pkg/api/claim_types.go

```diff
@@ -269,85 +299,96 @@ type ResourceClaimStatus struct {
+
+	// Setting this field is optional. If unset, the allocated devices are available everywhere.
+	//
+	// +optional
+	AvailableOnNodes *v1.NodeSelector `json:"availableOnNodes,omitempty" protobuf:"bytes,2,opt,name=availableOnNodes"`
+
+	// Shareable determines whether the claim supports more than one consumer at a time.
+	//
 	// +optional
 	Shareable bool `json:"shareable,omitempty" protobuf:"varint,3,opt,name=shareable"`
 }
 
 // DriverData holds information for processing by a specific kubelet plugin.
 type DriverData struct {
-	// DriverName specifies the name of the resource driver whose kubelet
+	// DriverName specifies the name of the DRA driver whose kubelet
 	// plugin should be invoked to process the allocation once the claim is
 	// needed on a node.
 	//
 	// Must be a DNS subdomain and should end with a DNS domain owned by the
 	// vendor of the driver.
 	DriverName string `json:"driverName" protobuf:"bytes,1,name=driverName"`
 
-	// Data contains all information about the allocation that the kubelet
-	// plugin will need.
-	//
-	// +optional
-	Data *StructuredDriverData `json:"data,omitempty" protobuf:"bytes,2,opt,name=data"`
-
-	// Alternative to "Data", not describe further here for the sake of brevity:
-	// OpaqueData *string // Set by control plane controller when using "classic DRA".
-}
-
-// StructuredDriverData is the in-tree representation of the allocation result.
-type StructuredDriverData struct {
 	// Config contains all the configuration pieces that apply to the entire claim
-	// and that were meant for the driver which handles these resources.
+	// and that were meant for the driver which handles these devices.
 	// They get collected during the allocation and stored here
 	// to ensure that they remain available while the claim is allocated.
 	//
 	// Entries are listed in the same order as in claim.config.
 	//
 	// +optional
 	Config []DriverConfigurationParameters `json:"config,omitempty"`
-
-	// Results lists all allocated driver resources.
-	//
-	// +listType=atomic
-	Results []RequestAllocationResult `json:"results" protobuf:"bytes,4,name=results"`
-}
-
-// DriverConfigurationParameters must have one and only one one field set.
-//
-// In contrast to ConfigurationParameters, the driver name is
-// not included and has to be infered from the context.
-type DriverConfigurationParameters struct {
-	Opaque *runtime.RawExtension `json:"opaque,omitempty" protobuf:"bytes,1,opt,name=opaque"`
 }
 
 // RequestAllocationResult contains configuration and the allocation result for
 // one request.
 type RequestAllocationResult struct {
+	// DriverName specifies the name of the DRA driver whose kubelet
+	// plugin should be invoked to process the allocation once the claim is
+	// needed on a node.
+	//
+	// Must be a DNS subdomain and should end with a DNS domain owned by the
+	// vendor of the driver.
+	DriverName string `json:"driverName" protobuf:"bytes,1,name=driverName"`
```

Comment:
Why do we need this both here and in `DriverData`? Speaking of which, I don't see `DriverData` embedded in any of the other types here. Remind me where it gets referenced again?

Reply:
`DriverName` became obsolete because I moved all of its fields directly into `AllocationResult`. I've removed it now.